chrome/content/coniks/context.js

```diff
@@ -0,0 +1,73 @@
+this.EXPORTED_SYMBOLS = ["AccountContext", "ConiksContext", "ContextsFile"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("chrome://otr/content/helpers.js");
+
+var OS;
+
+class ConiksContext {
+  constructor(opts) {
+    opts = opts || {};
+
+    if (opts.account === null) {
+      return null;
+    }
+    this.account = opts.account;
+    // user policies
+    this.allowUnsignedKeyChange = opts.allowUnsignedKeyChange === null || opts.allowUnsignedKeyChange === undefined ?
+                                  true : opts.allowUnsignedKeyChange;
+    this.allowPublicLookup = opts.allowPublicLookup === null || opts.allowPublicLookup === undefined ?
+                             true : opts.allowPublicLookup;
+  }
+
+  setPolicy(ctx) {
+    this.allowUnsignedKeyChange = ctx.allowUnsignedKeyChange;
+    this.allowPublicLookup = ctx.allowPublicLookup;
+  }
+}
+
+class AccountContext extends ConiksContext {
+  constructor(opts) {
+    opts = opts || {};
+
+    if (opts.account === null || opts.protocol === null) {
+      return null;
+    }
+    super(opts);
+    if (opts.fingerprint !== undefined || opts.fingerprint !== null) {
+      this.fingerprint = opts.fingerprint;
+    }
+    this.protocol = opts.protocol;
+    this.contacts = []; // array of ConiksContext
+    // this.TB = null;
+  }
+}
+
+var ContextsFile = {
+  path: helpers.profilePath("coniks.contexts"),
+
+  load: function(contexts) {
+    return helpers.fileExists(ContextsFile.path).then((exists) => {
+      if (exists) {
+        return helpers.readTextFile(ContextsFile.path)
+        .then(function(data) {
+          let ctxs = JSON.parse(data);
+          ctxs.forEach(function(c) {
+            let acc = `${c.account}@${c.protocol}`;
```

Comment:
Should use `accountname` (move to a helper) for this so that the format is guaranteed to stay the same.